cabby/geo/map_processing/map_structure.py

```diff
@@ -71,7 +71,7 @@ def __init__(
 
       osmid = self.nx_graph.nodes
       osmid_str = list(map(str, osmid))
-      nx.set_edge_attributes(self.nx_graph, osmid_str, 'osmid')
+      nx.set_node_attributes(self.nx_graph, osmid_str, 'osmid')
```

Comment:
This will create an `osmid` attribute for every node with the value equal to the _entire_ `osmid_str` list. Looking at the [docs for set_node_attributes](https://networkx.org/documentation/networkx-2.0/reference/generated/networkx.classes.function.set_node_attributes.html), I think you want `osmid_str` to be a dictionary mapping the node name to the string version of the osmid.

Reply:
Fixed